tests/pytests/test_multibyte_char_terms.py

```diff
@@ -0,0 +1,1436 @@
+    conn.execute_command('HSET', 'test:10', 't', 'ÙÛÜ')
+    conn.execute_command('HSET', 'test:11', 't', 'îï')
+    conn.execute_command('HSET', 'test:12', 't', 'ÎÏ')
+    conn.execute_command('HSET', 'test:13', 't', 'ôö')
+    conn.execute_command('HSET', 'test:14', 't', 'ÔÖ')
+    conn.execute_command('HSET', 'test:15', 't', 'ÿ')
+    conn.execute_command('HSET', 'test:16', 't', 'Ÿ')
+    conn.execute_command('HSET', 'test:17', 't', 'æ')
+    conn.execute_command('HSET', 'test:18', 't', 'Æ')
+
+    if not env.isCluster():
+        res = env.cmd(debug_cmd(), 'DUMP_TERMS', 'idx')
+        # only 9 terms are indexed, the lowercase representation of the terms
+        # with diacritics, but the diacritis are not removed.
+        env.assertEqual(len(res), 9)
+        env.assertEqual(res, ['àâä', 'æ', 'ç', 'éèêë', 'îï', 'ôö', 'ùûü', 'ÿ', 'œ'])
+
+def testDiacriticLimitation(env):
+    ''' Test that the diacritics are not removed, so the terms with diacritics
+    are not found when searching for terms without diacritics, and vice versa.
+    This limitation should be removed in the future, see MOD-5366.
+    '''
+    conn = getConnectionByEnv(env)
+    # In this test set the index language to FRENCH, because we want to
+    # search using stemmed words in french.
+    env.cmd('FT.CREATE', 'idx', 'ON', 'HASH',
+            'LANGUAGE', 'FRENCH', 'SCHEMA', 't', 'TEXT')
+
+    conn.execute_command('HSET', 'mot:1', 't', 'etude')
+    conn.execute_command('HSET', 'mot:2', 't', 'étude')
+    conn.execute_command('HSET', 'mot:3', 't', 'etudes')
+    conn.execute_command('HSET', 'mot:4', 't', 'études')
+
+    # the diacritics are not removed, so we got 6 different terms:
+    # the 4 original terms from the documents, and 2 stemmed terms.
+    if not env.isCluster():
+        res = env.cmd(debug_cmd(), 'DUMP_TERMS', 'idx')
+        expected = ['+etud', '+étud', 'etude', 'etudes', 'étude', 'études']
+        env.assertEqual(res, expected)
+
+    for dialect in range(1, 4):
+        run_command_on_all_shards(env, config_cmd(),
+                                  'SET', 'DEFAULT_DIALECT', dialect)
+
+        # search term without diacritics
+        # the diacritics are not removed, so the terms WITH diacritics are
+        # not found
+        expected = [2, 'mot:1', 'mot:3']
+        res = conn.execute_command(
+            'FT.SEARCH', 'idx', '@t:etude', 'NOCONTENT', 'SORTBY', 't')
+        env.assertEqual(res, expected, message=f'Dialect: {dialect}')
+        res = conn.execute_command(
+            'FT.SEARCH', 'idx', '@t:Etude', 'NOCONTENT', 'SORTBY', 't')
+        env.assertEqual(res, expected, message=f'Dialect: {dialect}')
+
+        # search term with diacritics
+        # the diacritics are not removed, so the terms WITHOUT diacritics are
+        # not found
+        expected = [2, 'mot:2', 'mot:4']
+        res = conn.execute_command(
+            'FT.SEARCH', 'idx', '@t:étude', 'NOCONTENT', 'SORTBY', 't',
+            'LANGUAGE', 'FRENCH')
```

Comment:
`LANGUAGE` `FRENCH` is not part of the original PR #5391 but has no effect here since it is the same language as in the index definition, e.g.,

> res = conn.execute_command(
            'FT.SEARCH', 'idx', '@t:étude', 'NOCONTENT', 'SORTBY', 't')
```